client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -133,8 +149,17 @@ const Checkout = React.createClass( {
 			return `/plans/${ this.props.sites.getSelectedSite().slug }/thank-you`;
 		}
 
-		// TODO: include the receipt ID in this URL when it is present
-		return `/checkout/${ this.props.sites.getSelectedSite().slug }/thank-you`;
+		if ( this.props.transaction.step.data && this.props.transaction.step.data.receipt_id ) {
+			receiptId = this.props.transaction.step.data.receipt_id;
+
+			this.props.fetchReceiptCompleted( receiptId, {
+				receiptId: receiptId,
+				purchases: this.getPurchasesFromReceipt()
+			} );
+		}
```

Comment:
I opened an issue for this: https://github.com/Automattic/wp-calypso/issues/3422